src/web/components/Donut.tsx

```diff
@@ -0,0 +1,189 @@
+import React from 'react';
+import { css } from 'emotion';
+
+import { headline } from '@guardian/src-foundations/typography';
+import { text } from '@guardian/src-foundations/palette';
+
+type Props = {
+    sections: Section[];
+    percentCutout?: number;
+    width?: number;
+    height?: number;
+};
+
+type Section = {
+    label: string;
+    value: number;
+    color: string;
+};
+
+const unitStyles = css`
+    ${headline.medium({ fontWeight: 'bold' })}
+    text-anchor: middle;
+`;
+
+const valueStyles = css`
+    ${headline.small({ fontWeight: 'bold' })}
+    fill: ${text.ctaPrimary};
+    text-anchor: middle;
+`;
+
+const labelStyles = css`
+    ${headline.xxsmall()}
+    fill: ${text.ctaPrimary};
+    text-anchor: middle;
+`;
+
+const withoutZeroSections = (sections: Section[]) =>
+    sections.filter(section => section.value !== 0);
+
+export const Donut = ({
+    sections,
+    percentCutout = 35,
+    width = 300,
+    height = 300,
+}: Props) => {
+    if (withoutZeroSections(sections).length === 1) {
+        console.warn('The Donut component requires at least 2 sections');
+        // TODO: Support showing 100% for a single section
+        return null;
+    }
+
+    if (sections.find(section => section.value === 0)) {
+        console.warn(
+            'Each section in the Donut component must have a non zero value',
+        );
+        // We handle these at the moment by filtering them out using withoutZeroSections()
+        // TODO: Support displaying 0% for sections where value is zero
+    }
+
+    const radius = Math.min(height / 2, width / 2);
+    const cutoutRadius = radius * (percentCutout / 100);
+
+    const totalValue = sections
+        .map(section => section.value)
+        .reduce((prev, next) => prev + next);
```

Comment:
```suggestion
    const totalValue = sections
        .reduce((acc, cur) => cur.value + acc, 0);
```

Reply:
I actually find the more verbose version easier to read and understand. But I agree `prev` and `next` is misleading - updated